### ISSUE TITLE
Streamlit 1.36.0 breaks functionality

### ISSUE THREAD

**Issue by danb27** [blackary/st_pages#99]
To reproduce: 

```
# ui.py
from st_pages import Page, show_pages

pages = [
    Page("ui.py", "home"),
    Page("page.py", "page"),
]
show_pages(pages)
```

```
# page.py
import streamlit as st

st.title("hi")
```

```
# requirements.txt
streamlit==1.36.0
st_pages==0.4.5
```

`streamlit run ui.py`

Results in exception:
```
AttributeError: 'str' object has no attribute 'main_script_path'
Traceback:
File "/Users/dan/repos/scrap/foo/.venv/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 589, in _run_script
    exec(code, module.__dict__)
File "/Users/dan/repos/scrap/foo/src/ui.py", line 7, in <module>
    show_pages(pages)
File "/Users/dan/repos/scrap/foo/.venv/lib/python3.11/site-packages/streamlit/runtime/metrics_util.py", line 408, in wrapped_func
    result = non_optional_func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/dan/repos/scrap/foo/.venv/lib/python3.11/site-packages/st_pages/__init__.py", line 252, in _show_pages
    rt._sources_watcher = LocalSourcesWatcher(rt._main_script_path)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/dan/repos/scrap/foo/.venv/lib/python3.11/site-packages/streamlit/watcher/local_sources_watcher.py", line 51, in __init__
    self._main_script_path = os.path.abspath(self._pages_manager.main_script_path)
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Temporary fix is to set `streamlit<1.36.0` in your application

**Comment by george-gca** [comment]
Just found out that [Streamlit 1.36.0](https://github.com/streamlit/streamlit/releases/tag/1.36.0) introduced [st.navigation and st.Page](https://docs.streamlit.io/develop/api-reference/navigation). Maybe now Streamlit officially supports this feature and this 3rd party is not necessary anymore?

**Comment by danb27** [comment]
> Just found out that [Streamlit 1.36.0](https://github.com/streamlit/streamlit/releases/tag/1.36.0) introduced [st.navigation and st.Page](https://docs.streamlit.io/develop/api-reference/navigation). Maybe now Streamlit officially supports this feature and this 3rd party is not necessary anymore?

Ooo interesting. I was just focused on fixing my build.

**Comment by andrew-weisman** [comment]
FWIW, I'm still interested in solving this within `st_pages` because the new functionality mentioned by @george-gca [does not allow for multiprocessing to be run in the app](https://github.com/streamlit/streamlit/issues/8957) (as far as I can tell).

**Comment by HoangNguyen689** [comment]
I am still facing this problem with show_pages_from_config().
Do you fix that in v0.5.0?

**Comment by blackary** [comment]
Yes, this should be fixed by release 0.5.0

**Comment by andrew-weisman** [comment]
Confirmed that 0.5.0 fixes the problem. Thanks so much @blackary!
